lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,23 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const prerelease = semver.prerelease(currentValue);
+    const snapshot = prerelease ? isSnapshot(prerelease) : false;
+    if (!prerelease || snapshot) {
```

Comment:
I don't understand why `!prerelease` is part of this. If it's not a prerelease then shouldn't the logic remain as it was before?

Reply:
As already mentioned here: https://github.com/renovatebot/renovate/pull/27274#discussion_r1492719144:
> A normal maven user would expect that bumping to a prerelease sets it to a SNAPSHOT version (https://github.com/renovatebot/renovate/pull/14763#discussion_r1000210402).
The else part is only for not normal (release or snapshot) maven versions, as it is hard to say what is expected in this case.

Reply:
Your logic is based on current version and not bump version

Reply:
The statement `!prerelease || snapshot` means "is normal maven version".
If this is the case the version bump can be made with the `inc` function of the semver library by setting the prerelease identifier to `SNAPSHOT` and prerelease identifier base to `false`. This should give the same result as for example the maven release plugin will do.
If the bump is not a `pre*` bump, the prerelease arguments do not have any effect in the `inc` function.